### DISCUSSION TITLE
How about modifying queryClient's staleTime to be settable?

### DISCUSSION THREAD

**Discussion by jakkku** [pmndrs/jotai#1526] [Ideas] [answered]
```tsx
const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      staleTime: Infinity,
    },
  },
})
export const App = () => {
  return (
    <QueryClientProvider client={queryClient}>
      <ReactQueryDevtools />
      <Provider initialValues={[[queryClientAtom, queryClient]] as const}>
        <App />
      </Provider>
    </QueryClientProvider>
  )
}
```
like above, if i want to set global options, `staleTime` is not settable.

https://github.com/pmndrs/jotai/blob/5b950ea38d0e4617c2a921f110160a97f6e8c782/src/query/atomWithQuery.ts#L85-L89

Looking at atomWithQuery, staleTime is fixed to 200.
Would it be a problem if I allowed the staleTime to be set? Is it okay if the minimum settable value is 200?

I made a simple modification and checked, the test seems to pass `staleTime: 200 ~ infinity`

**Comment by dai-shi** [comment, marked as answer]
We have moved to and encourage move to the new implementation [jotai-tanstack-query](https://github.com/jotai-labs/jotai-tanstack-query), which doesn't override `staleTime`. Please let us know if that works for you.

[The wrapper code](https://github.com/pmndrs/jotai/blob/5b950ea38d0e4617c2a921f110160a97f6e8c782/src/query/atomWithQuery.ts#L85-L89) is only to pass existing tests. `jotai/query` will be deprecated in the future.

**Reply by jakkku** [reply]
Oh, that's right. thank you 🙇‍♂️. This was my fault.
I was using `atomWithQuery` instead of `atomsWithQuery` in `'jotai-react-query'`.
Works well!!
[codesandbox](https://codesandbox.io/s/dawn-bird-99blcz?file=/src/App.tsx)
